### DISCUSSION TITLE
Is there anyway to access an internal repo?

### DISCUSSION THREAD

**Discussion by mattholy** [badges/shields#9319] [Q&A] [answered]
Is there anyway to access an internal repo by add an access token somewhere maybe?

The case is, I have a self-hosted gitlab instance, and all the repos are setup as internal or private.
I need a issues track badge, so what should I do?

**Comment by chris48s** [comment]
If you need to use shields badges with private repos or services that are behind a corporate firewall, the most common solution to this is to self-host your own instance of shields and use that rather than shields.io. There are docs on how to do this at https://github.com/badges/shields/blob/master/doc/self-hosting.md

**Reply by mattholy** [reply]
> If you need to use shields badges with private repos or services that are behind a corporate firewall, the most common solution to this is to self-host your own instance of shields and use that rather than shields.io. There are docs on how to do this at https://github.com/badges/shields/blob/master/doc/self-hosting.md

I had my own instance already. But the repos themselves are internal, like u need login to access.

**Reply by chris48s** [reply, marked as answer]
Then you should be able to set GitLab credentials using these config vars:

https://github.com/badges/shields/blob/master/doc/server-secrets.md#gitlab

Although do note that not 100% of GitLab badges currently work with auth
See https://github.com/badges/shields/issues/8768#issuecomment-1370159570 for details
